app/code/Magento/Sales/Model/AdminOrder/Create.php

```diff
@@ -886,11 +888,20 @@ public function moveQuoteItem($item, $moveTo, $qty)
                             );
                         }
 
-                        $cartItem = $cart->addProduct($product, $info);
-                        if (is_string($cartItem)) {
-                            throw new \Magento\Framework\Exception\LocalizedException(__($cartItem));
+                        $cartItems = $cart->getAllVisibleItems();
+                        $cartItemsToRestore = [];
+                        foreach ($cartItems as $value) {
+                            $cartItemsToRestore[$value->getData('item_id')] = $value->getData('item_id');
+                        }
```

Comment:
```suggestion
                        foreach ($cartItems as $cartItem) {
                            $cartItemsToRestore[$cartItem->getItemId()] = $cartItem->getItemId();
                        }
```

Could you please check this suggestion?